app/routes.py

```diff
@@ -1,2 +1,184 @@
+        return jsonify({"task":new_task.to_dict()}), 201
+
+@task_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_single_task(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+
+    #WV1: Get Task, Get Task Not Found
+    if request.method == "GET":
+        if task is None:
+            return make_response("", 404)
+        return jsonify({"task": task.to_dict()}), 200 
+
+    #WV1: Update Task, Update Task Not Found
+    #WV3: Update Task With Valid Completed At
+    elif request.method == "PUT":
+        input_data = request.get_json()
+        if task is None:
+            return make_response("", 404)
+        task.title = input_data["title"]
+        task.description = input_data["description"]
+        db.session.commit()
+        return ({"task":task.to_dict()}), 200
+
+    #WV1: Delete Task, Delete Task Not Found
+    elif request.method == "DELETE":
+        if task is None:
+            return make_response("", 404)
+        else:
+            db.session.delete(task)
+            db.session.commit()
+            return ({'details': f'Task {task_id} "{task.title}" successfully deleted'}), 200
+
+#WV3: Mark Comp on Incom, Mark Incom on Comp, Mark Comp on Comp, Marck Comp on Missing, Mark Incom on Missing 
+@task_bp.route("/<task_id>/<completion_status>", methods=["PATCH"])
+def mark_complete(task_id, completion_status):
+
+    task = Task.query.get(task_id)
+    PATH="https://slack.com/api/chat.postMessage"
+    if task is None:
+        return make_response("", 404)
+    
+    if completion_status == "mark_complete":
+        task.completed_at = date.today()
+
+        #WV4: Send Slack Message
+        token=os.environ.get("TOKEN")   
+        query_params = {
+            "token": token,
+            "channel": "task-notifications",
+            "text": f"Somone just completed task {task.title}"
+        }
+        requests.post(PATH, data=query_params)
+
+    if completion_status == "mark_incomplete":
+        task.completed_at = None
+    db.session.commit()
+    
+    return jsonify({"task": task.to_dict()})
+
+@goal_bp.route("", methods=["GET", "POST"])
+def handle_all_goals():
+    goal_response = []
+
+    #WV5: No Saved Goals, One Saved Goals
+    if request.method == "GET":
+        goals = Goal.query.all() 
+        for goal in goals:
+            goal_response.append(goal.to_dict())
+        return jsonify(goal_response), 200 
+
+    #WV5: Create Goal, Create Goal Missing Title
+    if request.method == "POST":
+        request_body = request.get_json()
+        try:
+            new_goal = Goal(title = request_body["title"])
+        except KeyError:
+            return ({'details': 'Invalid data'}, 400)       
+        db.session.add(new_goal)
+        db.session.commit()
+
+        return jsonify({"goal": new_goal.to_dict()}), 201 
+
+@goal_bp.route("/<goal_id>", methods=["GET", "DELETE", "PUT", "PATCH"])
+def handle_single_goal(goal_id):
+    goal_id = int(goal_id)
```

Comment:
Consider adding validation here so an error response is made if the user passes in a non-integer `goal_id`